System.IO.FileSystem/NativeFileStream.cs

```diff
@@ -0,0 +1,64 @@
+﻿//
+// Copyright (c) .NET Foundation and Contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System.Runtime.CompilerServices;
+
+namespace System.IO
+{
+    internal class NativeFileStream
+    {
+        // field is required for native interop
+#pragma warning disable IDE0051
+#pragma warning disable CS0169
+        object _fs;
+#pragma warning restore CS0169
+#pragma warning restore IDE0051
+
+        public const int TimeoutDefault = 0;
+        public const int BufferSizeDefault = 0;
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        public extern NativeFileStream(
+            string path,
+            int bufferSize);
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        public extern int Read(
+            byte[] buf,
+            int offset,
```

Comment:
Surely the offset will be greater than zero

Reply:
This is being called from `FileStream` which passes an `int` (comming from the full .NET API). Cant' convert without an explicit cast. Can't cast as it would potentially cause loos of data. 
In the end you can use the parameter to "go back" in the file (although not the usual usage) so I suppose that's the reason for the signed type.